### PR TITLE
windows-msi: ARM64 support

### PR DESCRIPTION
**Pull request by lstipakov**
Add building ARM64 MSI installer. This requires WiX 3.14.

ARM64 binaries are built by MSVC and expected to be found
at ../generic/image-arm64. If that directory is missing, ARM64
build variant is skipped.

ARM64 has following differences:

 - "checksum" header of binaries has value 0, so checksum validation is
 omitted

 - LZO library has a different name

 - Wintun feature is disabled, since ARM64 MSM for Wintun is not (yet)
available.

Signed-off-by: Lev Stipakov <lev@openvpn.net>

### CODE REVIEW

File: windows-msi/msi.wxs [hunk 1]
```diff
@@ -62,6 +62,11 @@
         <Property Id="DISABLEROLLBACK" Value="yes"/>
         <Property Id="EDITOR" Value="notepad.exe"/>
         <PropertyRef Id="NETFRAMEWORK40FULL"/>
+        <?if $(sys.BUILDARCH)~="x86" Or $(sys.BUILDARCH)~="x64" ?>
```
Comment (rozmansi) [inline windows-msi/msi.wxs:65]: Maybe we can drop binary checksum advertising completely? Some linkers set it, some don't. IIRC, files without checksum are always copied on repair. Savings are miniscule.

File: windows-msi/version.m4 [hunk 2]
```diff
@@ -5,12 +5,14 @@ dnl ============================================================
...
 define([PRODUCT_TAP_WIN_URL_x86],      [https://build.openvpn.net/downloads/releases/tap-windows-9.24.6-I601-i386.msm])
 define([PRODUCT_TAP_WIN_URL_amd64],    [https://build.openvpn.net/downloads/releases/tap-windows-9.24.6-I601-amd64.msm])
+define([PRODUCT_TAP_WIN_URL_arm64],    [https://build.openvpn.net/downloads/releases/tap-windows-9.24.6-I601-arm64.msm])
 define([PRODUCT_TAP_WIN_COMPONENT_ID], [tap0901])
 define([PRODUCT_TAP_WIN_NAME],         [TAP-Windows])
 
 dnl Wintun binaries
 define([PRODUCT_WINTUN_URL_x86],       [https://www.wintun.net/builds/wintun-x86-0.8.1.msm])
 define([PRODUCT_WINTUN_URL_amd64],     [https://www.wintun.net/builds/wintun-amd64-0.8.1.msm])
+define([PRODUCT_WINTUN_URL_arm64],     [https://www.wintun.net/builds/wintun-amd64-0.8.1.msm])
```
Comment (rozmansi) [inline windows-msi/version.m4:15]: You probably pointed this to x64 just to make build system happy to prepare prerequisites, regardless it does't need arm64 wintun.msm later anyway, right?

**Review by rozmansi — APPROVED** [review]
LGTM

File: windows-msi/msi.wxs (hunk 1 above)
Comment (lstipakov) [reply windows-msi/msi.wxs:65]: Agreed.

File: windows-msi/version.m4 (hunk 2 above)
Comment (lstipakov) [reply windows-msi/version.m4:15]: Right, I'll add a comment about it.

**Comment by lstipakov** [comment]
@rozmansi could you confirm that this looks good after changes (added comment + removed checksum) so that @cron2 could merge?

**Comment by rozmansi** [comment]
Cool. LGTM.

**Comment by cron2** [comment]
Hi,

On Wed, May 26, 2021 at 09:15:43AM -0700, Simon Rozman wrote:
> Cool. LGTM.

Is that an "ACK, please merge"?

gert
-- 
"If was one thing all people took for granted, was conviction that if you 
 feed honest figures into a computer, honest figures come out. Never doubted 
 it myself till I met a computer with a sense of humor."
                             Robert A. Heinlein, The Moon is a Harsh Mistress

Gert Doering - Munich, Germany                             ***@***.***

**Comment by rozmansi** [comment]
> Is that an "ACK, please merge"?

Yes, sorry.

Unfortunately, I don't have ARM64 computer to test it. Did a stare-review only.

**Comment by cron2** [comment]
Hi,

On Wed, May 26, 2021 at 09:29:19AM -0700, Simon Rozman wrote:
> Unfortunately, I don't have ARM64 computer to test it. Did a stare-review only.

I do the ARM tests :-) - Mac M1 with Parallels and Windows 10.

Windows is amazingly fast on that platform!

Stare-reviews by someone who understands MSI is exactly what is missing
in that constellation, so, thanks!

gert
-- 
"If was one thing all people took for granted, was conviction that if you 
 feed honest figures into a computer, honest figures come out. Never doubted 
 it myself till I met a computer with a sense of humor."
                             Robert A. Heinlein, The Moon is a Harsh Mistress

Gert Doering - Munich, Germany                             ***@***.***

**Comment by lstipakov** [comment]
The last change is to bundle `vcruntime140.dll` with all architectures, not only with ARM64. This is to enable MSVC release builds on x86/x64 and also good for consistency.

**Pull request merged**